### ISSUE TITLE
vimtex support

### ISSUE THREAD

**Issue by DerWeh** [hrsh7th/nvim-cmp#258]
Can you maybe help out, how to configure `nvim-cmp` to use  [vimtex](https://github.com/lervag/vimtex/) omnicompletion?

**Comment by hrsh7th** [comment]
You can use omni source.
https://github.com/hrsh7th/cmp-omni

NOTE: This source can be slow down your neovim.

```vim
autocmd FileType tex lua require('cmp').setup.buffer { sources = { { name = 'omni' } }  }`
```

**Comment by DerWeh** [comment]
Thanks a lot for the help, it seems I overlooked the `omni` source.

The problem how to trigger the `omni` source remains, after typing e.g. `\cite{` or `\cref{` I want to get completion.
E.g. in `deoplete` I could set
```viml
call deoplete#custom#var('omni', 'input_patterns', {
        \ 'tex': g:vimtex#re#deoplete
        \})
```
Previously, I hard-coded this like:
```viml
let g:deoplete#omni#input_patterns.tex = '\\(?:'
      \ .  '\w*cite\w*(?:\s*\[[^]]*\]){0,2}\s*{[^}]*'
      \ . '|\w*ref(?:\s*\{[^}]*|range\s*\{[^,}]*(?:}{)?)'
      \ . '|hyperref\s*\[[^]]*'
      \ . '|includegraphics\*?(?:\s*\[[^]]*\]){0,2}\s*\{[^}]*'
      \ . '|\w*(gls|Gls|GLS)(pl)?\w*(\s*\[[^]]*\]){0,2}\s*\{[^}]*'
      \ . '|includepdf(\s*\[[^]]*\])?\s*\{[^}]*'
      \ . '|includestandalone(\s*\[[^]]*\])?\s*\{[^}]*'
      \ . '|usepackage(\s*\[[^]]*\])?\s*\{[^}]*'
      \ . '|documentclass(\s*\[[^]]*\])?\s*\{[^}]*'
      \ . '|\w+'
      \ .')'
```
If we figure it out, it would of course be best to also update the `vimtex` manual, which contains the instructions for the deprecated `nvim-cmp` but not your new great plugin.

**Comment by hrsh7th** [comment]
@DerWeh Yes. vimtex provides `python regex` for completion. But it doesn't provide `vim regex`. So I can't support it.

You can specify keyword_pattern by the following codes.

```lua
cmp.setup {
  sources = {
    {
      name = 'omni',
      keyword_pattern = ... your pattern here (vim regex) ...
    }
  }
}
```

**Comment by Shougo** [comment]
You can use this instead.
https://github.com/lervag/vimtex/blob/master/autoload/vimtex/re.vim#L35

**Comment by ranebrown** [comment]
@storopoli Did you get this working? I have the below setup and I get completion for things like `\includegraphics` but `\cite{` doesn't show any completions without doing ctrl x ctrl o.

```vim
autocmd FileType tex lua require'cmp'.setup.buffer {
\   sources = {
\     { name = 'omni',
\       keyword_pattern = vim.g['vimtex#re#neocomplete']
\     },
\ }
```

**Comment by ranebrown** [comment]
I think that's what I'm using with `keyword_pattern = vim.g['vimtex#re#neocomplete']`. I get an error about invalid escape sequences if I try to copy in the full regex instead of using the variable.

**Comment by ranebrown** [comment]
I don't see anything about enabling the omni source?

@DerWeh Did you get a working `keyword_pattern`?

**Comment by donbex** [comment]
@ranebrown Looking at that configuration, I believe that @storopoli is getting completions from the TeXlab language server, rather than vimtex.

**Comment by storopoli** [comment]
Yes I am using TeXlab!

**Comment by Eternex49** [comment]
@ranebrown have you managed to find a solution?

**Comment by ranebrown** [comment]
I think it was resolved in the linked issue but I also mostly started using texlab lsp completions (also haven't written anything in latex in a while).
